src/sentry/monitors/models.py

```diff
@@ -312,6 +315,31 @@ class Meta:
     __repr__ = sane_repr("guid", "name")
 
 
+class MonitorEnvironmentManager(BaseManager):
+    """
+    A manager that consolidates logic for monitor enviroment updates
+    """
+
+    def ensure_environment(
+        self, monitor: Monitor, environment_name: str | None
+    ) -> MonitorEnvironment:
+        if not environment_name:
+            environment_name = "production"
+
+        # TODO: assume these objects exist once backfill is completed
```

Comment:
That's not guaranteed, since customers will be able to pass arbitrary envs right?

Reply:
Yep, you're right

Reply:
ah yeah we can revisit when we remove the dual write to make sure we're good on upsert